hw/ip/otbn/dv/uvm/otbn_sim_cfg.hjson

```diff
@@ -58,19 +58,29 @@ name:
   // TODO: This needs deciding upon. Eventually, it might be nice to allow this
   //       to be overridden on the command line ("I've got a bunch of ELF files
   //       that stress my new feature; please run the test suite with them")
-  otbn_elf_dir: "{sw_build_dir}"
-
+  otbn_elf_dir: "{run_dir}/build-out"
   run_opts: ["+otbn_elf_dir={otbn_elf_dir}"]
 
+  // OTBN run modes.
+  run_modes: [
+    {
+      name: build_otbn_rig_binaries_mode
+      pre_run_cmds: ["BUILD_ROOT={run_dir} cd {proj_root} && ./meson_init.sh",
+                     "{proj_root}/hw/ip/otbn/dv/uvm/gen-binaries.py --seed {seed} {otbn_elf_dir}"]
```

Comment:
If we are having to execute meson anyway, then a better approach would be to integrate the script itself into meson. If we choose to take that route, then #4289 should be of help.

Reply:
Hmm, do we actually need the `meson_init.sh` call? I would assume that we normally build binaries before we start running tests, so I wouldn't expect that to be part of the dvsim run.

Reply:
Be careful that `./meson_init.sh` can fail if it's re-run and the build directory exists.

Reply:
The build dir is set to `{run_dir}`, which is created brand new for every run.

Reply:
... which I think is problematic. Dvsim should not assume it is in full control of building all components of OpenTitan from source, but rely on the output from other build systems, outside of the dvsim boundary, instead. I have opened #4326 to discuss this further.

For this PR: @rswarbrick is gen-binaries.py able to cope with only $BUILD_ROOT being set, but not $OBJ_DIR, and will it set OBJ_DIR = $BUILD_ROOT/build-bin (an assumption that dvsim makes).

Reply:
No, but I can teach it that trick. I'll come back to this next week.